src/JustSaying/Messaging/MessageSerialization/SystemTextJsonSerializer`1.cs

```diff
@@ -0,0 +1,119 @@
+#if NET8_0_OR_GREATER
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using JustSaying.Extensions;
+using JustSaying.Messaging.MessageHandling;
+using JustSaying.Models;
+
+namespace JustSaying.Messaging.MessageSerialization;
+
+/// <summary>
+/// A class representing an implementation of <see cref="IMessageSerializer"/> for the <c>System.Text.Json</c> serializer.
+/// </summary>
+public class SystemTextJsonSerializer<T> : IMessageSerializer
+    where T : Message
+{
+    private readonly JsonSerializerOptions _options;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SystemTextJsonSerializer"/> class.
+    /// </summary>
+    public SystemTextJsonSerializer()
+        : this(null)
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SystemTextJsonSerializer"/> class.
+    /// </summary>
+    /// <param name="options">The optional <see cref="JsonSerializerOptions"/> to use.</param>
+    public SystemTextJsonSerializer(JsonSerializerOptions options)
+    {
+        if (options == null)
+        {
+            options = new JsonSerializerOptions
+            {
+#if NET8_0_OR_GREATER
+                DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
+#else
+                IgnoreNullValues = true,
+#endif
+            };
+        }
+
+        _options = options;
+    }
+
+    /// <inheritdoc />
+    public string GetMessageSubject(string sqsMessage)
+    {
+        using var body = JsonDocument.Parse(sqsMessage);
+        string subject = string.Empty;
+
+        if (body.RootElement.TryGetProperty("Subject", out var value))
+        {
+            subject = value.GetString() ?? string.Empty;
+        }
+
+        return subject;
+    }
+
+    public MessageAttributes GetMessageAttributes(string message)
+    {
+        var jsonDocument = JsonDocument.Parse(message);
+
+        if (!jsonDocument.RootElement.TryGetProperty("MessageAttributes", out var attributesElement))
+        {
+            return new MessageAttributes();
+        }
+
+        var attributes = new Dictionary<string, MessageAttributeValue>();
+        foreach(var obj in attributesElement.EnumerateObject())
+        {
+            var dataType = obj.Value.GetProperty("Type").GetString();
+            var dataValue = obj.Value.GetProperty("Value").GetString();
```

Comment:
We should make these defensive against the possibility of the properties not being there so we don't NRE.